.infra/workers.ts

```diff
@@ -1,295 +1,34 @@
-  },
-  {
-    topic: 'comment-upvoted',
-    subscription: 'api.comment-upvote-milestone-notification',
-  },
-  {
-    topic: 'api.v1.post-visible',
-    subscription: 'api.post-added-notification-v2',
-  },
-  {
-    topic: 'api.v1.member-joined-source',
-    subscription: 'api.member-joined-source-notification',
-  },
-  {
-    topic: 'api.v1.user-created',
-    subscription: 'api.user-created-cio',
-  },
-  {
-    topic: 'api.v1.banner-added',
-    subscription: 'api.banner-added',
-  },
-  {
-    topic: 'api.v1.banner-deleted',
-    subscription: 'api.banner-deleted',
-  },
-  {
-    topic: 'yggdrasil.v1.content-published',
-    subscription: 'api.content-published',
-  },
-  {
-    topic: 'api.v1.source-member-role-changed',
-    subscription: 'api.source-member-role-changed-notification',
-  },
-  {
-    topic: 'analytics-api.v1.experiment-allocated',
-    subscription: 'api.experiment-allocated',
-  },
-  {
-    topic: 'api.v1.source-created',
-    subscription: 'api.source-created-squad-user-action',
-  },
-  {
-    topic: 'api.v1.post-collection-updated',
-    subscription: 'api.post-collection-updated-notification',
-  },
-  {
-    topic: 'api.v1.user-readme-updated',
-    subscription: 'api.user-readme-images',
-  },
-  {
-    topic: 'api.v1.user-created',
-    subscription: 'api.user-created-personalized-digest-send-type',
-  },
-  {
-    topic: 'api.v1.comment-downvoted',
-    subscription: 'api.comment-downvoted-rep',
-  },
-  {
-    topic: 'api.v1.comment-downvote-canceled',
-    subscription: 'api.comment-downvote-canceled-rep',
-  },
-  {
-    topic: 'api.v1.squad-public-request',
-    subscription: 'api.v1.squad-public-request-notification',
-  },
-  {
-    topic: 'api.v1.user-streak-updated',
-    subscription: 'api.user-streak-updated-cio',
-  },
-];
-
-export const personalizedDigestWorkers = [
-  {
-    topic: 'api.v1.generate-personalized-digest',
-    subscription: 'api.personalized-digest-email',
-    args: {
-      ackDeadlineSeconds: 60,
-      deadLetterPolicy: {
-        deadLetterTopic: `projects/${gcp.config.project}/topics/${digestDeadLetter}`,
-        maxDeliveryAttempts: 5,
-      },
+const digestWorkersArgsMap: Record<string, WorkerArgs> = {
+  'api.personalized-digest-email': {
+    ackDeadlineSeconds: 60,
+    deadLetterPolicy: {
+      deadLetterTopic: `projects/${gcp.config.project}/topics/${digestDeadLetter}`,
+      maxDeliveryAttempts: 5,
     },
   },
-  {
-    topic: digestDeadLetter,
-    subscription: 'api.personalized-digest-email-dead-letter-log',
-    args: {
-      expirationPolicy: {
-        ttl: '',
-      },
+  'api.personalized-digest-email-dead-letter-log': {
+    expirationPolicy: {
+      ttl: '',
     },
   },
-];
+};
+
+export const personalizedDigestWorkers = commonDigestWorkers.map((worker) => {
+  const args: WorkerArgs = digestWorkersArgsMap[worker.subscription];
+
+  if (!args) {
+    return worker;
+  }
+
+  return { ...worker, ...args };
+});
```

Comment:
Can you please double check sure this merges correctly with what we had before? 🙏 Its pretty critical.

Reply:
Ah, I just test ran this, hence the merge issue, just so we can try this path out. I can clear it out now.

Reply:
Yes, I will double check to ensure things are still working as intended.